noodle-server/swagger/api.yaml

```
@@ -1,1023 +1,1876 @@
+        - in: path
+          name: courseId
+          schema:
+            type: integer
+          required: true
+      responses:
+        200:
+          description: Resource deleted successfully
+        400:
+          description: Bad Request
+        401:
+          description: No Access Permission to this resource
+        404:
+          description: No course found for this uid
+        500:
+          description: The server exploded while processing the request
+  /files:
+    get:
+      tags:
+        - files
+      summary: Get all files of the instance with their ids
+      responses:
+        200:
+          description: Successful
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FileCollection'
+        204:
+          description: No files found, but successful otherwise
+        400:
+          description: Bad Request
+        404:
+          description: Resource not found, something has gone wrong
+        500:
+          description: Internal Server Error
+        401:
+          description: Unauthorized
+    post:
+        tags:
+          - files
+        summary: Upload a new file, and obtain its new uid
+        requestBody:
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/FileCreationRequest"
+        responses:
+          200:
+            description: Successful operation
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/File'
+          400:
+            description: Bad Request
+          401:
+            description: No Access Permission to this resource
+          404:
+            description: Something was not found
+          500:
+            description: The server exploded while processing the request
+  /file/{fileUid}:
+    get:
+      tags:
+        - files
+      summary: Get a file by uid and its contents
+      parameters:
+        - in: path
+          name: fileUid
+          schema:
+            type: integer
+          required: true
+      responses:
+        200:
+          description: Successful Operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/File"
+        400:
+          description: Bad Request
+        401:
+          description: No Access Permission to this resource
+        404:
+          description: No file found for this uid
+        500:
+          description: The server exploded while processing the request
+    put:
+      tags:
+        - files
+      summary: Replace a file, without altering the uid
+      parameters:
+        - in: path
+          name: fileUid
+          schema:
+            type: integer
+          required: true
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/File'
```

```suggestion
              $ref: '#/components/schemas/FileCreationRequest'
```
Hier brauchen wir die zusätzlichen Daten von `File` nicht. `last_updated` wird vom Server bestimmt und `uid` ist bereits im Url-Pfad.